proto/cmp/services/notification/v2/notify.proto

```diff
@@ -0,0 +1,119 @@
+  rpc CancellationAcceptedNotification(CancellationAccepted) returns (google.protobuf.Empty);
+  rpc CancellationAcceptedByTheOwnerNotification(CancellationAcceptedByTheOwner) returns (google.protobuf.Empty);
+  rpc CancellationCounteredNotification(CancellationCountered) returns (google.protobuf.Empty);
+  rpc CancellationWithdrawnNotification(CancellationWithdrawn) returns (google.protobuf.Empty);
+  rpc CancellationRejectedNotification(CancellationRejected) returns (google.protobuf.Empty);
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event TokenBought(uint256 indexed tokenId, address indexed buyer);
+// ```
+message TokenBought {
+  uint64 token_id = 1;
+  string tx_id = 2;
+  cmp.types.v1.UUID mint_id = 3;
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event TokenExpired(uint256 indexed tokenId);
+// ```
+message TokenExpired {
+  uint64 token_id = 1;
+  cmp.types.v1.UUID mint_id = 2;
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event CancellationPending(uint256 indexed tokenId, address indexed proposedBy, uint256 refundAmount, uint16 cancellationReason, uint16 cancellationReasonVersion);
+// ```
+message CancellationPending {
+  uint64 token_id = 1;
+  string proposed_by = 2;
+  uint64 refund_amount = 3;
+  cmp.services.cancellation.v1.CancellationReason reason = 4;
+  string tx_id = 5;
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event CancellationAccepted(uint256 indexed tokenId, address indexed acceptedBy, uint256 refundAmount);
+// ```
+message CancellationAccepted {
+  uint64 token_id = 1;
+  string accepted_by = 2;
+  uint64 refund_amount = 3;
+  string tx_id = 4;
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event CancellationAcceptedByTheOwner(uint256 indexed tokenId, address indexed acceptedBy, uint256 refundAmount);
+// ```
+message CancellationAcceptedByTheOwner {
+  uint64 token_id = 1;
+  string accepted_by = 2;
+  uint64 refund_amount = 3;
+  string tx_id = 4;
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event CancellationCountered(uint256 indexed tokenId, address indexed counteredBy, uint256 newRefundAmount);
+// ```
+message CancellationCountered {
+  uint64 token_id = 1;
+  string countered_by = 2;
+  uint64 new_refund_amount = 3;
+  string tx_id = 4;
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event CancellationWithdrawn(uint256 indexed tokenId, address indexed cancelledBy, uint16 reason, uint16 reasonVersion);
+// ```
+message CancellationWithdrawn {
+  uint64 token_id = 1;
+  string cancelled_by = 2;
+  cmp.services.cancellation.v1.WithdrawalReason reason = 3;
+  string tx_id = 4;
+}
+
+// Related on-chain event:
+//
+// ```solidity
+// event CancellationRejected(uint256 indexed tokenId, address indexed rejectedBy, uint16 rejectionReason, uint16 rejectionReasonVersion);
+// ```
+message CancellationRejected {
+  uint64 token_id = 1;
+  string rejected_by = 2;
+  cmp.services.cancellation.v1.RejectionReason reason = 3;
+  string tx_id = 4;
```

Comment:
We're using a lot of times a transaction id (tx_id, transaction_id) just as a string and I think it would be better to wrap this into a dedicated type (for proper documentation of what this actually is and as a preparation for proto-validate rules for this field)

Reply:
> Agreed - Doesn't hurt to have it defined.

Added.

> We're using a lot of times a transaction id (tx_id, transaction_id) just as a string and I think it would be better to wrap this into a dedicated type

I will do these in another PR as it will introduce a lot of changes (because of the automatic version bumping script)